### PR TITLE
release: qase-python-commons 3.0.2b5

### PR DESCRIPTION
**Pull request by gibiw**
Fixed the following issues with network profiler:
- ignore requests to `qase.io` host
- check that `body` and `headers` attributes are in the request
- fix ignoring response for `urllib3` package
- add a creation step for redirected requests

### CODE REVIEW

File: qase-python-commons/src/qase/commons/profilers/network.py [hunk 1]
```diff
@@ -47,15 +47,17 @@ def wrapper(self, request, *args, **kwargs):
     def _urllib3_request_wrapper(self, func):
         @wraps(func)
         def wrapper(self, method, url, *args, **kwargs):
+            if 'qase.io' in url:
+                return func(self, method, url, *args, **kwargs)
+
```
Comment (NickVolynkin) [inline qase-python-commons/src/qase/commons/profilers/network.py:52]: So, the profiler won't profile requests to qase.io, made for creating test runs and reporting results, right?

Users can use a proxy or custom domain for Qase app. Let's bind this to the Qase URL config variable.

File: qase-python-commons/src/qase/commons/profilers/network.py [hunk 2]
```diff
@@ -68,15 +70,25 @@ def _log_pre_request(request):
...
                 request_url=request.url,
-                request_body=request.body,
-                request_headers=request.headers,
+                request_body=request.body if hasattr(request, 'body') else None,
+                request_headers=request.headers if hasattr(request, 'headers') else None
             ),
         )
 
     @staticmethod
     def _log_post_response(response, url=None, *args, **kwargs):
```
Comment (NickVolynkin) [inline qase-python-commons/src/qase/commons/profilers/network.py:79]: post means the HTTP POST method or "post response" as "after response"?

File: qase-python-commons/src/qase/commons/profilers/network.py [hunk 3]
```diff
@@ -68,15 +70,25 @@ def _log_pre_request(request):
...
         status = response.status if hasattr(response, 'status') else response.status_code
         profiler = NetworkProfilerSingleton.get_instance()
+        if profiler.step is None:
+            profiler.step = Step(
+                id=str(uuid.uuid4()),
+                step_type=StepType.REQUEST,
+                data=StepRequestData(
```
Comment (NickVolynkin) [inline qase-python-commons/src/qase/commons/profilers/network.py:86]: Will request and response be reported as two adjacent steps or a single step?

Comment (gibiw) [reply qase-python-commons/src/qase/commons/profilers/network.py:86]: in a single step

File: qase-python-commons/src/qase/commons/profilers/network.py (hunk 2 above)
Comment (gibiw) [reply qase-python-commons/src/qase/commons/profilers/network.py:79]: it is like "after response". I will rename the method

File: qase-python-commons/src/qase/commons/profilers/network.py (hunk 1 above)
Comment (gibiw) [reply qase-python-commons/src/qase/commons/profilers/network.py]: fixed

File: qase-python-commons/src/qase/commons/profilers/network.py (hunk 2 above)
Comment (gibiw) [reply qase-python-commons/src/qase/commons/profilers/network.py]: fixed

**Review by NickVolynkin — APPROVED** [review]

**Pull request merged**